lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -2306,7 +2306,7 @@ public function getDefaultValueDeclarationSQL($field)
             return " DEFAULT '" . $this->convertBooleans($default) . "'";
         }
 
-        return " DEFAULT '" . $default . "'";
+        return " DEFAULT " . $this->quoteDefaultStringLiteral($default);
```

Comment:
Given that `quoteStringLiteral()` only escapes quotes and backslashes by means of `str_replace()`, it may not work with encodings where the `0x27` byte (the single quote in ASCII) is part of a multi-byte sequence.

This approach to escaping is used everywhere on the platform level, so it'd be good to hear from other maintainers whether this approach is valid in general.

/cc @Ocramius @lcobucci

Reply:
This seems to only affect schema definitions, so I'm not too worried about it for now. It is a problem if people rely on it for anything runtime-related.